packages/ui/src/SelectField.tsx

```diff
@@ -154,6 +174,43 @@ const getMenuContainer = (menuPortalTarget: 'body' | 'self' | HTMLElement | null
   return menuPortalTarget
 }
 
+type GetSelectedOptionFromValueProps<V> = {
+  value: V | V[] | null
+  options: SelectFieldOption<V>[]
+  isMulti: boolean
+}
+
+/**
+ * Transforms the value passed to our Select component so that it's in a format as underlying react-select expects
+ * @param value - single value or an array of values
+ * @param options - an array of objects describing all possible options with their labels
+ * @param isMulti - boolean indicating whether we're in a multiple mode or not
+ */
+export function getSelectedOptionFromValue<V>({ value, options, isMulti }: GetSelectedOptionFromValueProps<V>) {
+  if (isMultipleModeGuard(value, isMulti)) {
+    // if it's multi value, let's transform a value array to an array containing SelectFieldOptions
+    return (
+      value?.map((val) => ({
+        value: val,
+        // we need to find instead of value: object mapping so that we can support objects as values
+        label: (options.find(({ value }) => value === val) as SelectFieldOption<V>)?.label ?? val,
+      })) ?? []
+    )
```

Comment:
```suggestion
    return value.map((val) => ({
      value: val,
      // we need to find instead of value: object mapping so that we can support objects as values
      label: (options.find(({ value }) => value === val) as SelectFieldOption<V>)?.label ?? val,
    }))
```

Reply:
@danad02 but value can be `null` if you use multi mode combined with creatable

Reply:
@jvorcak It can't, I just tested it

Reply:
@jvorcak Confirming that it's not an issue related to this PR. We have wrong types for `multi` from the beginning.

Reply:
<img width="674" alt="Screenshot 2021-03-04 at 9 40 50" src="https://user-images.githubusercontent.com/1083817/109935292-badb3480-7ccd-11eb-8f2e-b73a353895b2.png">
<img width="737" alt="Screenshot 2021-03-04 at 9 40 32" src="https://user-images.githubusercontent.com/1083817/109935309-be6ebb80-7ccd-11eb-884c-dfec1b0602a9.png">

Reply:
Well yes, you got it as a default value in the `useState`. If we change typings and you use `[]` as the default value, you won't get null there.

Reply:
@jvorcak Sorry to drop this non-related issue on you during this great enhancement. I just think it's a good thing to fix it sooner rather than later.